### ISSUE TITLE
how to avoid launching an autorun opencl kernel?

### ISSUE THREAD

**Issue by ronghongbo** [halide/Halide#3076]
Hello,

I'm generating an Altera OpenCL file from Halide. This file may contain some "autorun" kernels, which the host does not need to launch at all. But now we see these kernels are still launched, and we got a CL_INVALID_KERNEL_NAME error in halide_opencl_run(). 

Any idea how we may make Halide not launch these kernels?

Thanks,
Hongbo

**Comment by zvookin** [comment]
I assume one is modifying at least the runtime if not the compiler to add the autorun attribute. I expect the host code generation will need to be modified to not call halide_opencl_run on the kernels it is marking as autorun.

**Comment by ronghongbo** [comment]
I agree. But we cannot find anywhere halide_opencl_run is generated in Halide.

**Comment by zvookin** [comment]
See:
    https://github.com/halide/Halide/blob/master/src/CodeGen_GPU_Host.cpp#L491

**Comment by ronghongbo** [comment]
Great! Thanks.

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the
> reason below. The reasoning was checked against the current source, and a second pass
> tried and failed to refute it, but no human verified this specific text.
> **If the reason is wrong, please reopen the issue and say so** — some fraction of these
> are expected to be wrong.

**Reason (CLOSE_STALE):** A support question about an out-of-tree Altera/Intel FPGA OpenCL fork that adds an 'autorun' kernel attribute. zvookin pointed at the halide_opencl_run call site in CodeGen_GPU_Host and the reporter replied 'Great! Thanks.' Nothing was ever requested of upstream Halide.

**Evidence:** Question answered in-thread and acknowledged 2018-06-22; the call site still exists (src/CodeGen_GPU_Host.cpp emits the halide_*_run call). No upstream 'autorun' concept exists in src/.

Closing - this was a support question about an out-of-tree Altera OpenCL fork, answered in the thread (the launch is emitted from CodeGen_GPU_Host's For visitor). No upstream change was requested.
